src/libraries/System.Text.Json/src/System/Text/Json/AppContextSwitchHelper.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Text.Json
+{
+    internal static class AppContextSwitchHelper
+    {
+        private const string SourceGenReflectionFallbackEnabled_SwitchName = "System.Text.Json.Serialization.EnableSourceGenReflectionFallback";
+        private static int s_SourceGenReflectionFallbackEnabled_CachedValue;
+
+        public static bool IsSourceGenReflectionFallbackEnabled
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            get => GetCachedSwitchValue(SourceGenReflectionFallbackEnabled_SwitchName, ref s_SourceGenReflectionFallbackEnabled_CachedValue, defaultValue: false);
```

Comment:
I don't understand why this needs to be so complicated. It would be easier to read, and I believe work better with Tiered compilation, if it was written like:

https://github.com/dotnet/runtime/blob/20dd6ea921028b3a2cc660654b65112077671fb5/src/libraries/Common/src/System/SR.cs#L10

Reply:
Code was taken from https://github.com/dotnet/runtime/blob/9028dc2d715a827a21a5ffbd029bc7a24073406b/src/libraries/Common/src/System/LocalAppContextSwitches.Common.cs, but in hindsight most of the functionality is not needed. Simplifying makes sense.